src/components/hooks/use-cookie-state.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react'
+import { CookieStorage } from 'common/storage'
+import { isNullUndefined, isJSONString } from 'common/utility'
+
+export const useCookieState = (defaultValue, key, options) => {
+    const cookie_state = new CookieStorage(key)
+    const [value, setValue] = useState(() => {
+        const sticky_value = cookie_state.get(key)
+        const result = sticky_value
+            ? isJSONString(sticky_value)
+                ? JSON.parse(sticky_value)
+                : sticky_value
+            : defaultValue
```

Comment:
```suggestion
        const result = isJSONString(sticky_value) ? JSON.parse(sticky_value) : defaultValue
```
🚧 Comment #1
Do we need to check for falsy here if we already check it in `isJSONString` function? I think we should do all the parsing in a method that accepts two arguments, maybe something like:

`parseJSONString(value, default_value)`

parseJSON if succeed return parsed json else return `default_value`

Reply:
the idea was to parse sticky value if it is available, if not then use default value, however, there's possibility that sticky value is not a JSON (normal string). its a good idea to use parseJSONString function however, not with 2 arguments for a fallback, something more isolated will be like

```
const parseJSONString = (value) => isJSONString(value) ? JSON.parse(value) : value

const result = sticky_value ? parseJSONString(sticky_value) : defaultValue
```

Reply:
the reason why not 2 arguments:
1. that one function will have too many operations
2. when reading the name of parseJSONString, normally people will expect it to parse `a` value (one argument), its hard to realize it will have more than one argument unless if you look back again to the function definition

Reply:
I see, that is a legit reasoning 👍🏼